### PR TITLE
Mosip 12805

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/PauseFlowPredicate.java [hunk 1]
```diff
@@ -54,20 +53,15 @@ public boolean matches(Exchange exchange) {
 
 		String message = (String) exchange.getMessage().getBody();
 		JsonObject json = new JsonObject(message);
-		JsonObject tags = json.getJsonObject("tags");
+
 		LOGGER.info(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/PauseFlowPredicate.java]: can we make this debug log ?

File: registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java [hunk 2]
```diff
@@ -1,185 +1,184 @@
...
+		exchange = new DefaultExchange(endpoint);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testRoutePredicatePositive() throws Exception {
+		MessageDTO messageDTO = new MessageDTO();
+		messageDTO.setRid("10002100741000120201231071308");
+		Map<String, String> tags = new HashMap<>();
+		tags.put("META_INFO-META_DATA-machineId", "10074");
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java]: Lets use the same HOTLISTED tag for all the test as before and change the matchExpression accordingly

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/RegistrationTransactionTypeCode.java [hunk 3]
```diff
@@ -1,47 +1,56 @@
...
+	QUALITY_CHECK,
+
+	/** The biometric Authentication. */
+	BIOMETRIC_AUTHENTICATION,
+
+	/** The securezone notification stage. */
+	SECUREZONE_NOTIFICATION,
+
+	/** The workflow event update. */
+	WORKFLOW_EVENT_UPDATE,
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/RegistrationTransactionTypeCode.java]: WORKFLOW_EVENT_UPDATE and WORKFLOW_ACTION_SERVICE is not a transaction type

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java [hunk 4]
```diff
@@ -596,37 +422,36 @@ private void sendMessage(MessageDTO object, MessageBusAddress address) {
...
-			throws JsonParseException, JsonMappingException, ApisResourceAccessException, JsonProcessingException,
-			PacketManagerException, com.fasterxml.jackson.core.JsonProcessingException, IOException,
-			WorkflowActionException {
+	private void removeHotlistedTag(InternalRegistrationStatusDto internalRegistrationStatusDto)
+			throws ApisResourceAccessException, JsonProcessingException, PacketManagerException
+			{
+		String resumeRemoveTags = internalRegistrationStatusDto.getResumeRemoveTags();
+		if (StringUtils.isEmpty(resumeRemoveTags))
+			return;
+		String[] resumeRemoveTagsArray = resumeRemoveTags.split(",");
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java]: Fixed: trim the content in resumeRemoveTagsArray

File: registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowActionApi.java [hunk 5]
```diff
@@ -128,16 +149,42 @@ public void processURL(RoutingContext ctx) {
 			regProcLogger.debug("WorkflowActionApi:processURL called for registration ids {}",
 					workflowIds);
 			validator.validate(workflowActionDTO);
-				workflowActionService.processWorkflowAction(workflowIds,
-					workflowActionDTO.getRequest().getWorkflowAction());
+			List<InternalRegistrationStatusDto> internalRegistrationStatusDtos = new ArrayList<InternalRegistrationStatusDto>();
+			for (String workflowId : workflowIds) {
+				InternalRegistrationStatusDto internalRegistrationStatusDto = registrationStatusService
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowActionApi.java]: Fixed : queried all workflowids in a singlecall

File: registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowEventUpdateVerticle.java [hunk 6]
```diff
@@ -122,15 +124,17 @@ public MessageDTO process(MessageDTO object) {
 		registrationStatusDto.setStatusComment(workflowEventDto.getStatusComment());
 		registrationStatusDto.setDefaultResumeAction(workflowEventDto.getDefaultResumeAction());
 		LocalDateTime resumeTimeStamp = DateUtils.parseToLocalDateTime(workflowEventDto.getResumeTimestamp());
-		LocalDateTime updateTimeStamp = DateUtils.parseToLocalDateTime(workflowEventDto.getEventTimestamp());
 		registrationStatusDto.setResumeTimeStamp(resumeTimeStamp);
-		registrationStatusDto.setUpdateDateTime(updateTimeStamp);
 		registrationStatusDto.setUpdatedBy(USER);
+			registrationStatusDto.setResumeRemoveTags(workflowEventDto.getResumeRemoveTags());
+			registrationStatusDto
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowEventUpdateVerticle.java:130]: Fixed : moved code to method

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/PauseFlowPredicate.java (hunk 1 above)
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/PauseFlowPredicate.java]: Fixed : updated to debug

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/RegistrationTransactionTypeCode.java (hunk 3 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/RegistrationTransactionTypeCode.java]: updated to WORKFLOW_PAUSE and WORKFLOW_RESUME

File: registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java (hunk 2 above)
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java]: Fixed test cases

File: registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowActionApi.java [hunk 7]
```diff
@@ -128,16 +149,42 @@ public void processURL(RoutingContext ctx) {
...
 					workflowIds);
 			validator.validate(workflowActionDTO);
-				workflowActionService.processWorkflowAction(workflowIds,
-					workflowActionDTO.getRequest().getWorkflowAction());
+			List<InternalRegistrationStatusDto> internalRegistrationStatusDtos = new ArrayList<InternalRegistrationStatusDto>();
+			for (String workflowId : workflowIds) {
+				InternalRegistrationStatusDto internalRegistrationStatusDto = registrationStatusService
+						.getRegistrationStatus(workflowId);
+				String user="";
+				if (Objects.nonNull(ctx.user())
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowActionApi.java]: moved the code to separate method

File: registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java [hunk 8]
```diff
@@ -437,8 +437,14 @@ private void removeHotlistedTag(InternalRegistrationStatusDto internalRegistrati
...
-		String[] resumeRemoveTagsArray = resumeRemoveTags.split(",");
 		List<String> deleteTags = new ArrayList<String>();
+		String[] resumeRemoveTagsArray = resumeRemoveTags.split(",");
+		if (resumeRemoveTagsArray.length > 0) {
+			for (int i = 0; i < resumeRemoveTagsArray.length; i++) {
+				deleteTags.add(resumeRemoveTagsArray[0].trim());
+			}
+		}
+
 		deleteTags.addAll(Arrays.asList(resumeRemoveTagsArray));
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java]: I think this line needs to be removed

File: registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java [hunk 9]
```diff
@@ -106,14 +106,14 @@ public void testRoutePredicatePositive() throws Exception {
...
-		tags.put("META_INFO-META_DATA-machineId", "10074");
+		tags.put("HOTLISTED", "operator");
 		messageDTO.setTags(tags);
 		exchange.getMessage().setBody(objectMapper.writeValueAsString(messageDTO));
 		LocalDateTime dateTimeBefore = DateUtils.getUTCCurrentDateTime().plusSeconds(432000);
 		assertTrue(pauseFlowPredicate.matches(exchange));
 		LocalDateTime dateTimeAfter = DateUtils.getUTCCurrentDateTime().plusSeconds(432000);
 		JsonObject json = new JsonObject((String) exchange.getMessage().getBody());
-		assertEquals("ResumeFromBeginning", json.getString("defaultResumeAction"));
+		assertEquals("STOP_PROCESSING", json.getString("defaultResumeAction"));
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java:116]: in match expression above stop processing is in small case, here it is caps, is that ok ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java (hunk 8 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowActionService.java]: Fixed

File: registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java (hunk 9 above)
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java:116]: Fixed

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**